frontend/app/templates/components/encryptable/form.hbs

```diff
@@ -1,107 +1,277 @@
+                        {{this.userService.username}}
+                      {{else}}
+                        {{team.name}}
+                      {{/if}}
+                    </el.control>
+                  </form.element>
+                </div>
+                <div class="form-group col-md-12" id="folder-power-select">
+                  <form.element id="encryptable-form-folder" if="encryptable-form-folder"
+                                @options={{this.availableFolders}} @label={{t "folder"}} @controlType="power-select"
+                                @value={{this.changeset.folder}}
+                                @searchEnabled={{true}} @allowClear={{true}} @searchField={{"name"}}
+                                @customError={{t (validation-error-key this.changeset.error.folder.validation)}} as
+                                |el|>
+                    <el.control class={{if this.changeset.error.folder.validation "invalid-input-ps"}}
+                                @disabled={{this.isFolderDropdownDisabled}}
+                                @placeholder={{t "encryptable/credentials.edit.folder_placeholder"}}
+                                @renderInPlace={{true}} @onChange={{this.setFolder}} as |folder|>
+                      {{folder.name}}
+                    </el.control>
+                  </form.element>
+                </div>
               </div>
             </div>
-
-            <div class="col-md-6 order-3 order-md-4">
-              <div class="col-sm-12 secret">
-                <form.element @label={{t "password"}} id="cleartext-password" class={{if
-                        this.changeset.error.cleartextPassword.validation "invalid-input-name"}}
-                              @property="cleartextPassword" @customError={{t
-                        (validation-error-key this.changeset.error.cleartextPassword.validation)}} as |el|>
-                  <el.control autocomplete="off" tabindex="3" @name="cleartextPassword" ></el.control>
-                </form.element>
-                <PasswordStrengthMeter class="col-12 col-lg-9 px-0 mb-2 mb-md-0" @password={{this.changeset.cleartextPassword}}/>
-              </div>
+            <form.group class="col-md-12">
+              <form.element id="encryptable-form-description" @label={{t "description"}} @property="description"
+                            @controlType="textarea"
+                            class="{{if
+                              this.changeset.error.description.validation "invalid-input"}} vertical-resize"
+                            customError={{t
+                (validation-error-key this.changeset.error.description.validation)}} as |el|>
+                <el.control tabindex="7"></el.control>
+              </form.element>
+            </form.group>
+          </div>
+          <div class="encryptable-form-attributes">
+            <div class="row">
+              {{#if this.isUsernameFieldActive}}
```

Comment:
When we extract the field into a separate components, there is a compiler error for all properties set on form.element, like @property, @customError or @label. This is the case because the form.element isn't in the same component as the opening and closing tag of BsForm.

The BsForm basically opens and closes the whole form and with it the possibility to use form.group and form.element. I also tried passing down the form into the new child component for the fields and use it like that, but since we can't use mustaches in element tag names this approach also doesn't work.

I saw other people having the same issues and they got the recommendation to leave the form in one solid component, and that it doesn't really make sense to split it into multiple child components.

Maybe you have a solution, or you think it is fine when we leave it as it is

Reply:
A different approach to make the code more DRY and reduce the length of the file would be to remove the booleans for the fields completely and add a second array for all active items, so we can switch the currently active fields from an array called activeFields to an array called inactiveFields.

With that we could iterate over it in the component and load in the fields one by one.

Advantages to current form:
- more DRY
- less file length
- when adding a new field, it will always display at the end of the form, e.g. currently when only having password field active and adding the usename field, or whatever field that comes before password in out template, it adds the username before the password field and moves every following field in the form, which is kinda confusing. With the new way with the arrays the new element will be placed as the last element.

Disadvantages:
- Template could get more complex, since we still need if statements inside the each loop of the password field and custom attrs field.

For me this approach would be the best solution, since we make it more user friendly and the file will be much shorter. The complexity we add is basically already in the template right now, just implemented differently.

I suggested the approach to @Robin481 and he told me to ask you @mtnstar, what you think about this way of approaching this feedback.